test/jasmine/tests/hover_label_test.js

```diff
@@ -0,0 +1,245 @@
+var d3 = require('d3');
+
+var Plotly = require('@src/index');
+var Fx = require('@src/plots/cartesian/graph_interact');
+var Lib = require('@src/lib');
+
+var createGraphDiv = require('../assets/create_graph_div');
+var destroyGraphDiv = require('../assets/destroy_graph_div');
+
+describe('hover info', function() {
+    'use strict';
+
+    var mock = require('@mocks/14.json'),
+        evt = {
+                clientX: mock.layout.width/ 2,
+                clientY: mock.layout.height / 2
+        };
+
+    afterEach(destroyGraphDiv);
+
+    describe('hover info', function() {
+        var mockCopy = Lib.extendDeep({}, mock);
+
+        beforeEach(function(done) {
+            Plotly.plot(createGraphDiv(), mockCopy.data, mockCopy.layout).then(done);
```

Comment:
It looks like we could have 1 beforeEach and multiple `it`s?
pseudocode
```javascript

var mockData, mockLayout

describe('hover tests', function () {
   mockData = mockData()
   mockLayout = mockLayout()
   
   beforeEach( function (done) {
       Plotly.plot(gd, mockData, mockLayout).then(done)
   })
  
   it('responds to hover', function () { // test that uses mockData and mockLayout }
   it('resonds to hover x', function () { // test ... }
})
```

We can put the var declarations global scope and then assign fresh copies in the single BeforeEach. No need for multiple describes and beforeEach's. If that is dooable is should really clear up the test.

For comparison check out the many Jest (Jasmine fork) tests in Filewell. We have a set of patterns that might be useful. https://github.com/plotly/streambed/blob/master/shelly/filewell/static/filewell/src/stores/__tests__/DirectoryStore-test.js

Here there is one BeforeEach to refresh state before each test and a few describe blocks that break the test into logical chunks. If we are just testing hover here one describe is probably enough

Check around the src tree, every folder has a __test__ folder with Jest tests.